node/operations.js

```diff
@@ -85,9 +87,25 @@ OperationTombstone.prototype.extendLogInfo = function extendLogInfo(info) {
     return info;
 };
 
+OperationTombstone.prototype.destroy = function destroy(now) {
+    var self = this;
+
+    self.destroyed = true;
+
+    self.onTimeout(now);
+};
+
 OperationTombstone.prototype.onTimeout = function onTimeout(now) {
     var self = this;
 
+    if (!self.destroyed && now < self.timeout + self.time) {
```

Comment:
Hmm, if they truly are timing out early, that's a bug in the time heap.  Can you please pull out this "fix" into a separate PR from the instrumentation changes?  I'd then rather try to see if we can fix the heap bug (or prove me wrong in labelling it such), but this kind of logic (and its corollary below) is exactly the sort of thing we're supposed to have encapsulated away behind the time heap.